lib/Doctrine/DBAL/Query/QueryBuilder.php

```diff
@@ -465,6 +466,27 @@ public function select($select = null) : self
         return $this->add('select', $selects);
     }
 
+    /**
+     * Adds a DISTINCT flag to this query.
+     *
+     * <code>
+     *     $qb = $conn->createQueryBuilder()
+     *         ->select('u.id')
+     *         ->distinct()
+     *         ->from('users', 'u');
+     * </code>
+     *
+     * @param bool $flag If true, then DISTINCT is added to the query
+     *
+     * @return $this This QueryBuilder instance.
+     */
+    public function distinct(bool $flag = true)
```

Comment:
```suggestion
    final public function distinct(bool $flag = true) : self
```

Reply:
Agree with `self` part, but not sure of `final` specificator. It would be the only one `final` method in QueryBuilder.

Reply:
Yeah but it is one of the most recent ones, right? We can wait for the advice of another maintainer if you want, I will not merge this without the approval of some other maintainers anyway.